GUI.py

```diff
@@ -0,0 +1,647 @@
+
+    def init_frame(self) -> None:
+        # Sizers
+        self.mainSizer = wx.BoxSizer(wx.HORIZONTAL)
+        self.graphSizer = wx.BoxSizer()
+        self.controlSizer = wx.BoxSizer(wx.VERTICAL)
+
+        # Panels
+        self.graph = MyPanel(self)
+        self.graph.draw_graph()
+        self.graphs = MyPanel(self)
+        self.graphs.draw_graphs()
+        self.graphs.Hide()
+        self.control = MyPanel(self)
+        self.control.show_buttons()
+        self.control.graph = self.graph
+        self.control.graphs = self.graphs
+
+        #
+        self.graphSizer.Add(self.graph, 0, wx.ALL, 5)
+        self.graphSizer.Add(self.graphs, 0, wx.ALL, 5)
+        self.controlSizer.Add(self.control, 0, wx.ALL, 5)
+        self.mainSizer.Add(self.graphSizer, 0, wx.LEFT, 5)
+        self.mainSizer.Add(self.controlSizer, 0, wx.RIGHT, 5)
+        self.SetSizer(self.mainSizer)
+        self.Fit()
+
+
+class MyPanel(wx.Panel):
+    """Custom panel class."""
+
+    def __init__(self, parent: MyFrame) -> None:
+        super(MyPanel, self).__init__(parent=parent)
+
+    def show_buttons(self) -> None:
+        hbox = wx.BoxSizer(wx.VERTICAL)
+        self.btn = wx.Button(self, label="Open Text File")
+        self.btn.Bind(wx.EVT_BUTTON, self.on_open)
+
+        # -----------------
+        self.cwd = os.path.join(
+            os.path.abspath(os.path.dirname(__file__)), "Refraction_index"
+        )
+
+        refl_files = get_files_list(self.cwd)
+        self.wvlngh, self.n_coef = refection_coef_read(
+            os.path.join(self.cwd, refl_files[1])
+        )
+        cb = wx.ComboBox(
+            self, choices=refl_files, style=wx.CB_READONLY, value=refl_files[1]
+        )
+        cb.Bind(wx.EVT_COMBOBOX, self.on_select)
+        self.st = wx.StaticText(self, label="")
+        self.st.SetLabel("Refractive index file:")
+        # -----------------
+
+        self.auto = wx.Button(self, label="Auto GaAs depth")
+        self.auto.Enable(False)
+        self.auto.Bind(wx.EVT_BUTTON, self.calc_auto)
+        # ------------------
+
+        self.calc_btn = wx.Button(self, label="Calculate depth")
+        self.calc_btn.Enable(False)
+        self.calc_btn.Bind(wx.EVT_BUTTON, self.calc_choice)
+        # ------------------
+
+        self.chk_box = wx.CheckBox(self, label="Full Report")
+
+        hbox.Add(self.btn, 0, wx.CENTER, 5)
+        hbox.Add(self.auto, 0, wx.CENTER, 5)
+        hbox.Add(self.calc_btn, 0, wx.CENTER | wx.BOTTOM, 10)
+        hbox.Add(self.st, 0, wx.EXPAND | wx.ALIGN_CENTER_HORIZONTAL | wx.TOP, 5)
+        hbox.Add(cb, 0, wx.EXPAND | wx.ALIGN_CENTER_HORIZONTAL | wx.ALL, 1)
+        hbox.Add(self.chk_box, 0, wx.EXPAND | wx.ALIGN_CENTER_HORIZONTAL | wx.TOP, 5)
+
+        self.SetSizer(hbox)
+
+    def on_select(self, event: wx.Event) -> None:
+        i = event.GetString()
+        self.wvlngh, self.n_coef = refection_coef_read(os.path.join(self.cwd, i))
+
+    def on_open(self, event: wx.Event) -> None:
```

Comment:
event - not used parameter

Reply:
No, this function is event-dependent: you press the button, and the code runs. So 2 positional arguments are given to this function.